File: GOLD/milkmeasurement.cpp
#include <iostream>
#include <vector>
#include <set>
#include <algorithm>
#include <climits>

using namespace std;

int main() {

    freopen("measurement.in", "r", stdin);
    freopen("measurement.out", "w", stdout);
    int n;
    cin >> n;

    vector<int> cows(3, 0);
    vector<pair<int,pair<int,int>>> v(n);

    for (int i = 0 ; i < n ; i++) {
        int a, b, c;
        string s;
        cin >> a >> s >> c;
        if (s == "Bessie") {
            b = 0;
        } else if (s == "Elsie") {
            b = 1;
        } else {
            b = 2;
        }
        v[i] = {a, {b, c}};
    }

    sort(v.begin(), v.end());

    set<int> prev;
    prev.insert(0);
    prev.insert(1);
    prev.insert(2);

    int ans = 0;
    for (int i = 0 ; i < n ; i++) {
        cows[v[i].second.first] += v[i].second.second;
       //cout << v[i].second.first << " " << v[i].second.second << endl;

        int curmax = -INT_MAX;
        set<int> cur;
        for (int j = 0 ; j < 3 ; j++) {
            if (cows[j] > curmax) {
                cur.clear();
                cur.insert(j);
            } else if (cows[j] == curmax) {
                cur.insert(j);
            }
            curmax = max(curmax, cows[j]);
        }
        if (prev != cur) ans++;
        //cout << endl;
        prev = cur;
    }
    cout << ans << endl;
}

File: GOLD/milkmeasurement.py

def getKey(x):
    return x[0]


infile = open("measurement.in","r")
outfile = open("measurement.out","w")

cows = [7,7,7]
n = int(infile.readline())

times = []

for x in range(n):
    line = infile.readline().split(" ")
    a = int(line[0])
    c = int(line[2])
    b = 0
    if (line[1] == "Bessie"): 
        b = 0
    elif (line[1] == "Elsie"): 
        b = 1
    else:
        b = 2
    times.append([a,b,c])

times = sorted(times, key=getKey)


prev = set()
prev.add(0)
prev.add(1)
prev.add(2)


ans = 0
for x in times:
    cows[x[1]] += x[2]

    curmax = -1000000
    cur = set()
    for y in range(3):
        if (cows[y] >= curmax):

            if (cows[y] > curmax):
                cur.clear()
            cur.add(y)

            curmax = cows[y]
    
    if (prev != cur):
        ans+=1
    prev = cur

outfile.write(str(ans))

